packages/contracts-bedrock/contracts/universal/CrossDomainMessenger.sol

```diff
@@ -345,17 +350,32 @@ abstract contract CrossDomainMessenger is
             "CrossDomainMessenger: message has already been relayed"
         );
 
+        // If there is not enough gas left to perform the external call and finish the execution,
+        // return early and assign the message to the failedMessages mapping.
         // If `xDomainMsgSender` is not the default L2 sender, this function
         // is being re-entered. This marks the message as failed to allow it
         // to be replayed.
-        if (xDomainMsgSender != Constants.DEFAULT_L2_SENDER) {
+        if (
+            !SafeCall.hasMinGas(_minGasLimit, RELAY_RESERVED_GAS) ||
```

Comment:
There's a lot of stuff happening between this check and the CALL...

Reply:
Including the xDomainMessageSender  SSTORE.

Reply:
I believe the `RELAY_RESERVED_GAS` includes this plus the cost of execution after the `CALL` subcontext ends.

Reply:
That's probably a constraint that should be explicitly stated if it's not already

Reply:
Think this can be resolved given the discussion today and the outline in [XDM-Bricked-Withdrawals-Fix](https://www.notion.so/oplabs/XDM-Bricked-Withdrawals-Fix-b738c44a5e6047a687cfc796511108b4?pvs=4).

It would be nice to explicitly break up the implicit assumption that the `5,700` additional gas buffer from the `hasMinGas` check's `40,000` buffer covers these instructions. So it's explicit + clear going forward.

Reply:
I've added an explicit buffer within `baseGas` as well as the call to `hasMinGas`